dvc/command/git_hook.py

```diff
@@ -1,36 +1,37 @@
-import logging
-import os
-
 from dvc.command.base import CmdBaseNoRepo, fix_subparsers
-from dvc.exceptions import NotDvcRepoError
-
-logger = logging.getLogger(__name__)
 
 
 class CmdHookBase(CmdBaseNoRepo):
+    cmd = None
+    collect_analytics = False
+
     def run(self):
+        from dvc.exceptions import NotDvcRepoError
+        from dvc.main import main
         from dvc.repo import Repo
 
+        assert self.cmd
         try:
             repo = Repo()
             repo.close()
         except NotDvcRepoError:
             return 0
 
-        return self._run()
+        return main([self.cmd], disable_analytics=True)
```

Comment:
Push/status/etc should trigger the analytics. The fact that it is slow means that we need to look deeper into it and optimize it.

Reply:
Network connection do take some time.

Reply:
@skshetry But that's why we launch it in a separate process. And that matters during regular `dvc` operation too, not only for hooks. There might be some regression.

Reply:
Nope, not a regression. `pre-commit` waits for all of the processes to finish.

Reply:
@skshetry We launch analytics in a detached state, so it shouldn't wait for that process to finish. Unless there is something wrong on our side, of course.